### PR TITLE
[CIFuzz] Adding dry_run mode to check_build

### PR DESCRIPTION
**Pull request by Leo-Neat**
Prevents crash from surfacing when dry run mode is enabled and check fuzzers fails.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
Did someone ask for this? I think it's not terrible to have this for consistency, supporting weird usecases but check_build is simple enough that I'm not sure anyone would need this.

File: infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def main():
   out_dir = os.path.join(workspace, 'out')
   if cifuzz.check_fuzzer_build(out_dir):
     return 0
-  return 1
+  return error_code
```
Comment (jonathanmetzman) [inline infra/cifuzz/actions/build_fuzzers/build_fuzzers_entrypoint.py]: "error_code" should probably be called "returncode"
(You can use an underscore, subprocess doesn't though).

**Comment by Leo-Neat** [comment]
> Did someone ask for this? I think it's not terrible to have this for consistency, supporting weird usecases but check_build is simple enough that I'm not sure anyone would need this.

@Dor1s Mentioned it in #3419. I think it is a good idea for consistency.

**Comment by Dor1s** [comment]
Yeah, if someone prefers dry_run mode, IMO they shouldn't get any failure

**Pull request merged**